src/lib/y2network/config_reader/autoyast.rb

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2network/interface"
+require "y2network/config"
+require "y2network/config_reader/autoyast_routing"
+require "y2network/autoinst_profile/networking_section"
+
+Yast.import "Lan"
+
+module Y2Network
+  module ConfigReader
+    # This class is responsible of importing Autoyast configuration
+    class Autoyast
+      # @return [AutoinstProfile::NetworkingSection]
+      attr_reader :section
+
+      # Constructor
+      #
+      # @param section [AutoinstProfile::NetworkingSection]
+      def initialize(section)
+        @section = section
+      end
+
+      # @return [Y2Network::Config] Network configuration
+      def config
+        interfaces = find_interfaces
+        routing = section.routing ? AutoyastRouting.new(section.routing).config : nil
```

Comment:
@imobachgs @jreidinger we started a discussion about the routing config specially when importing in Autoyast whithout reading the system. If the routing section is empty shall we instantiate a Y2Network::Routing config with defaults or just use `nil` as no routing section has been imported. In other case when exporting it will be different but somehow expected.

Reply:
Adding the `nil` initialization means we will need to handle it correctly when writing, exporting .. etc

Reply:
Perhaps I would rename this class to `AutoinstRouting` (like `AutoinstProfile`, `AutoinstProposal`, etc.).

Reply:
good question as I think we face it in many modules. Sadly I do not see yet good solution for this issue. I vote for defaults to make our code easier and result more predictable.

Reply:
Used `Autoinst` and `AutoinstRouting`